data/exporter.go

```diff
@@ -0,0 +1,199 @@
+				logger.WithError(err).Error("failed to get/create file: %w", err)
+			}
+
+			select {
+			case req.err <- err: // does not block because it is buffered (1) channel, but better safe than sorry
+			case <-ctx.Done():
+				logger.WithError(ctx.Err()).Error("failed to export request")
+			}
+		case <-ticker.C:
+			worker.closeIdleFiles()
+		case <-ctx.Done():
+			worker.closeIdleFiles()
+			return
+		}
+	}
+}
+
+func (s *ExportService) SubmitGetPayloadRequest(ctx context.Context, gpr structs.SignedBlindedBeaconBlock) error {
+	request := exportRequest{dt: BlockBidAndTraceData, data: gpr.Raw(), slot: gpr.Slot(), timestamp: time.Now(), id: gpr.BlockHash().String(), err: make(chan error, 1)}
+
+	// submit request
+	select {
+	case s.requests <- request:
+	case <-ctx.Done():
+		return ctx.Err()
+	}
+
+	// wait for response
+	select {
+	case err := <-request.err:
+		return err
+	case <-ctx.Done():
+		return ctx.Err()
+	}
+}
+
+func writeToFile(file *os.File, req exportRequest) error {
+	// Encode struct as JSON and write to base64 encoder
+	_, err := file.Write([]byte(strconv.Itoa(req.timestamp.Nanosecond())))
+	if err != nil {
+		return fmt.Errorf("failed to write data timestamp: %w", err)
+	}
+
+	_, err = file.Write([]byte(";"))
+	if err != nil {
+		return fmt.Errorf("failed to write data sep ';': %w", err)
+	}
+
+	_, err = file.Write([]byte(req.id))
+	if err != nil {
+		return fmt.Errorf("failed to write data identifier: %w", err)
+	}
+
+	_, err = file.Write([]byte(";"))
+	if err != nil {
+		return fmt.Errorf("failed to write data sep ';': %w", err)
+	}
+
+	// Create base64 encoder that writes directly to gzip writer
+	encoder := base64.NewEncoder(base64.StdEncoding, file)
+	encoder.Close()
+
+	// Create gzip writer that writes directly to file
+	writer := gzip.NewWriter(encoder)
+	defer writer.Close()
+
+	_, err = writer.Write(req.data)
+	if err != nil {
+		return fmt.Errorf("failed to write data: %w", err)
+	}
+
+	if err := writer.Flush(); err != nil {
+		return fmt.Errorf("failed to flush writer: %w", err)
+	}
+
+	// Flush any remaining data from the encoder and writer
+	if err := encoder.Close(); err != nil {
+		return fmt.Errorf("failed to close encoder: %w", err)
+	}
+
+	_, err = file.Write([]byte("\n"))
+	if err != nil {
+		return fmt.Errorf("failed to write new data sep newline': %w", err)
+	}
+
+	return nil
+}
+
+type worker struct {
+	id      int
+	datadir string
+	files   map[string]fileWithTimestamp
+
+	logger log.Logger
+}
+
+func newWorker(id int, datadir string, logger log.Logger) *worker {
+	return &worker{id: id, datadir: datadir, files: make(map[string]fileWithTimestamp), logger: logger}
+}
+
+func (w *worker) getOrCreateFile(req exportRequest) (*os.File, error) {
+	// get
+	filename := fmt.Sprintf("%s/%s/output_%d_%d.json", w.datadir, toString(req.dt), req.slot, w.id)
+	fileWithTs, ok := w.files[filename]
+	if ok {
+		fileWithTs.ts = time.Now()
+		return fileWithTs.File, nil
+	}
+
+	// create
+	file, err := os.Create(filename)
```

Comment:
Create truncates file if already exists. That would mean it after the restart it would erase all the data we have for such slot/worker pair. You need to check if the file exists and open that or create the new one. Also for opening file you probably like to use os.O_APPEND